csrc/alias_analysis.h

```diff
@@ -107,10 +103,20 @@ class AliasAnalysisResult {
 //
 // Therefore, I chose to run alias analysis both before segmentation and in
 // schedulers. The former, used by MarkAliasesPreparePass, updates layouts to
-// enable aliases; the latter, used by NoOpScheduler, calls
+// enable aliases; the latter, used by ExprEvalScheduler, calls
 // Fusion::aliasOutputToInput to mark aliases.
+enum class EmptyAllocationAs {
+  kCanonical,
+  kUndetermined,
+};
+
+// Finds aliases of the fusion inputs. The analysis should be conservative --
+// when the analysis says B is an alias of input A and that B's preferred layout
+// is compliant with the required layout, `ExpressionEvaluator::evaluate(B)`
+// should produce an `at::Tensor` that's an alias of the `at::Tensor` bound to
+// A.
 AliasAnalysisResult findAliases(
     Fusion* fusion,
-    bool can_override_empty_allocation_domain = true);
```

Comment:
As a refactor, I changed this bool to an enum.

Reply:
What is your motivation behind switching it to `enum` -- more error-proof, easier to refactor the code etc?
I found it easier to understand what the boolean was just from the name as opposed to the enum.

Reply:
Readability, type safety and extensibility. I failed to find a reference in Google's cpp style guide, but I found plenty of posts talking about boolean argument vs enum. 

Do you find the name `EmptyAllocationAs::...` confusing? I'm happy to hear your suggestion on that.

Reply:
Got it. `kCanonical` was not clear to me, maybe `kLogical`?. `kUndetermined` is clear about what it means.